src/components/MailberryPopupOption.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useContext, useEffect, useRef, useState } from "react";
+import { FormPopupOptions, formPopupOptions } from "../types";
+import { getSubscriptionFromLocalStorage } from "../utils/localStorage";
+import { FormContext } from "./MailberryForm";
+
+type MailberryFormPopupProps = {
+  href: string;
+  signature: boolean;
+  thanksMessage: string;
+  handleSubmit: (e: React.FormEvent<HTMLFormElement>) => void;
+  showAt: FormPopupOptions;
+  formId: string;
+  children: React.ReactNode;
+}
+
+const MailberryFormPopup = ({ href, signature, thanksMessage, handleSubmit, showAt, formId, children }: MailberryFormPopupProps) => {
+  const { isSubmitted, isSubmitting, showErrorMessage, showThanksMessage } = useContext(FormContext);
+  const [showOverlay, setShowOverlay] = useState(false);
+  const formContainerRef = useRef<HTMLDivElement>(null);
+
+  const checkScrollPosition = () => {
+    const percent = 0.3;
+    const scrollY = window.scrollY;
+    const fullHeight = document.documentElement.scrollHeight;
+    const windowHeight = window.innerHeight;
+    const thirtyPercentOfPageview = fullHeight * percent;
+
+    if (scrollY + windowHeight >= thirtyPercentOfPageview) {
+      !showOverlay && setShowOverlay(true)
+    }
+
+    window.removeEventListener('scroll', checkScrollPosition);
+    return
+  };
+
+  // Dismiss overlay when click outside the form, only usefull for popup format
+  const handleDismissOverlayWhenClickOutside = (event: MouseEvent) => {
+    const target = event.target as HTMLElement;
+
+    if (formContainerRef.current && !formContainerRef.current.contains(target)) {
+      setShowOverlay(false);
+    }
+  };
+
+  // Dismiss overlay when is displayed, only usefull for popup format
+  const handleDismissOverlay = () => {
+    setShowOverlay(false);
+
+    const alreadySubscribed=localStorage.getItem(`subscribed_${formId}`);
+    if(!alreadySubscribed){
+      localStorage.setItem(`closed_${formId}`, Date.now().toString());
+    }
+  };
+
+  // Dismiss overlay when click outside the form
+  useEffect(() => {
+    document.addEventListener("mousedown", handleDismissOverlayWhenClickOutside);
+
+    return () => {
+      document.removeEventListener("mousedown", handleDismissOverlayWhenClickOutside);
+    };
+  }, []);
+
+  // Set popup option
+  useEffect(() => {
+    if(showAt === formPopupOptions.AT_30_PERCENT_OF_PAGEVIEW){

```

Comment:
instead of receiving a constant like `AT_30_PERCENT_OF_PAGEVIEW` make it receive the percentage, then build the snippet accordingly from the mailberry_client.